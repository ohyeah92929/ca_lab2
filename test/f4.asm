.orig x3000
and r0, r0, #0 ; 0x5020, r0: 0
and r1, r1, r0 ; 0x5240, r1: 0
lea r4, bye; 0xe104, r4: x300c(0011000000001010)
jmp r4; 0xc100
add r0, r0, #12; 0x102C
add r0, r0, #12; 0x102C
bye add r0, r0, #12; 0x102C, r0: 12
add r1, r0, r0; 0x1200, r1: 24
rshfl r5, r4, 1; 0xd711, r5: x1805(0001100000000101)
not r6, r4; 0x9d3f, r6: xcff5(1100111111110101)
rshfa r2, r6, 1; 0xd5b1 r2: xe7fa (1110011111111010)
xor r3, r2, r6; 0x9686, r3: x280f(0010100000001111)
halt
lab .fill x2222;
.end