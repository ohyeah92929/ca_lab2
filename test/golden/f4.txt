
Memory content [0x3000..0x301c] :
-------------------------------------
 0x3000 (12288) : 0x5020
 0x3002 (12290) : 0x5240
 0x3004 (12292) : 0xe403
 0x3006 (12294) : 0xc100
 0x3008 (12296) : 0x102C
 0x300a (12298) : 0x102C
 0x300c (12300) : 0x102C
 0x300e (12302) : 0x1200
 0x3010 (12304) : 0x0000
 0x3012 (12306) : 0x0000
 0x3014 (12308) : 0x0000
 0x3016 (12310) : 0x0000
 0x3018 (12312) : 0x0000
 0x301a (12314) : 0x0000
 0x301c (12316) : 0x0000


Current register/bus values :
-------------------------------------
Instruction Count : 11
PC                : 0x0000
CCs: N = 0  Z = 2  P = 9
Registers:
0: 0x000c
1: 0x0018
2: 0x0000
3: 0x280f
4: 0x300c
5: 0x0000
6: 0x0000
7: 0x300e

